generator/graphql-kotlin-federation/src/main/kotlin/com/expediagroup/graphql/generator/federation/execution/EntitiesDataFetcher.kt

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2022 Expedia, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.expediagroup.graphql.generator.federation.execution
+
+import com.expediagroup.graphql.generator.federation.exception.InvalidFederatedRequest
+import com.expediagroup.graphql.generator.federation.execution.resolverexecutor.FederatedTypePromiseResolverExecutor
+import com.expediagroup.graphql.generator.federation.execution.resolverexecutor.FederatedTypeResolverExecutor
+import com.expediagroup.graphql.generator.federation.execution.resolverexecutor.ResolvableEntity
+import com.expediagroup.graphql.generator.federation.extensions.toDataFetcherResult
+import graphql.execution.DataFetcherResult
+import graphql.schema.DataFetcher
+import graphql.schema.DataFetchingEnvironment
+import java.util.concurrent.CompletableFuture
+
+private const val TYPENAME_FIELD = "__typename"
+private const val REPRESENTATIONS = "representations"
+
+/**
+ * Federated _entities field data fetcher.
+ */
+open class EntitiesDataFetcher(
```

Comment:
This would be a breaking change so we would need another major. Do you need to change the name? It looks like the type if the class is the same

Reply:
i think is worth it, we are pursuing a signature similar to other subgraph implementations